compiler/src/iree/compiler/Codegen/Common/IREEExpandStridedMetadata.cpp

```diff
@@ -236,17 +236,24 @@ struct IREEExpandStridedMetadataPass
 };
 } // namespace
 
+void populateIREEExpandExtractStridedMetadataPatterns(
+    RewritePatternSet &patterns) {
+  memref::populateExpandStridedMetadataPatterns(patterns);
+  patterns.add<ResolveExtractMetadataFromHalInterfaceBindingSubspan>(
+      patterns.getContext());
+}
+
 void populateIREEResolveExtractStridedMetadataPatterns(
-    MLIRContext *context, RewritePatternSet &patterns) {
+    RewritePatternSet &patterns) {
   memref::populateResolveExtractStridedMetadataPatterns(patterns);
-  patterns.insert<ResolveExtractMetadataFromHalInterfaceBindingSubspan>(
-      context);
+  patterns.add<ResolveExtractMetadataFromHalInterfaceBindingSubspan>(
+      patterns.getContext());
 }
 
 void IREEExpandStridedMetadataPass::runOnOperation() {
   MLIRContext *context = &getContext();
   RewritePatternSet patterns(context);
-  populateIREEResolveExtractStridedMetadataPatterns(context, patterns);
+  populateIREEExpandExtractStridedMetadataPatterns(patterns);
```

Comment:
Can you give me an example of why this is needed? If we need this please add it in the `populateIREEResolveExtractStridedMetadataPatterns`.